integrations/kube-agent-updater/pkg/maintenance/basichttp.go

```diff
@@ -0,0 +1,133 @@
+// represents a critical update.
+func (b *basicHTTPMaintenanceClient) Get(ctx context.Context) (bool, error) {
+	versionURL := b.baseURL.JoinPath(maintenancePath)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, versionURL.String(), nil)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	res, err := b.client.Do(req)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return false, trace.Errorf("non-200 status code received: '%d'", res.StatusCode)
+	}
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	payload := strings.TrimSpace(string(body))
+	// Validating early that the payload can be converted to a boolean allows to
+	// gracefully catch connectivity error caused by mitm infrastructure such as
+	// corporate proxies.
+	result, err := stringToBool(payload)
+	return result, trace.Wrap(err)
+}
+
+// BasicHTTPMaintenanceTrigger gets the critical status from an HTTP response
+// containing only a truthy or falsy string.
+// This is used typically to trigger emergency maintenances from a file hosted
+// in a s3 bucket or raw file served over HTTP.
+// BasicHTTPMaintenanceTrigger uses basicHTTPMaintenanceClient and wraps it in a cache
+// in order to mitigate the impact of too frequent reconciliations.
+// The structure implements the maintenance.Trigger interface.
+type BasicHTTPMaintenanceTrigger struct {
+	name         string
+	cachedGetter func(context.Context) (bool, error)
+}
+
+// Name implements maintenance.Triggernd returns the trigger name for logging
+// and debugging pursposes.
+func (g BasicHTTPMaintenanceTrigger) Name() string {
+	return g.name
+}
+
+// Default returns what to do if the trigger can't be evaluated.
+// BasicHTTPMaintenanceTrigger should fail open, so the function returns true.
+func (g BasicHTTPMaintenanceTrigger) Default() bool {
+	return false
+}
+
+// CanStart implements maintenance.Trigger
+func (g BasicHTTPMaintenanceTrigger) CanStart(ctx context.Context, _ v1.Object) (bool, error) {
+	result, err := g.cachedGetter(ctx)
+	return result, trace.Wrap(err)
+}
+
+// NewBasicHTTPMaintenanceTrigger builds and return a Trigger checking a public HTTP endpoint.
+func NewBasicHTTPMaintenanceTrigger(name string, baseURL *url.URL) Trigger {
+	client := &http.Client{
+		Timeout: httpTimeout,
+	}
+	httpMaintenanceClient := &basicHTTPMaintenanceClient{
+		baseURL: baseURL,
+		client:  client,
+	}
+
+	return BasicHTTPMaintenanceTrigger{name, cache.NewTimedMemoize[bool](httpMaintenanceClient.Get, cacheDuration).Get}
+}
+
+func stringToBool(input string) (bool, error) {
+	switch {
+	case strings.EqualFold("true", input), strings.EqualFold("yes", input):
+		return true, nil
+	case strings.EqualFold("false", input), strings.EqualFold("no", input):
+		return false, nil
+	default:
+		return false, trace.BadParameter("cannot convert input to boolean: %s", input)
+	}
+}
```

Comment:
Can we just use `strconv.ParseBool` from stdlib?

Reply:
I was originally going to suggest this, but `strconv.ParseBool` doesn't support "yes"/"no" and I was unsure if those were expected valid values.

Reply:
[According to the RFD](https://github.com/gravitational/teleport/pull/20622/files#diff-f63de43bebaee990731ad22fb63fd0a07156af3858506b3110839113e4328bc5R125-R133) we must support "yes".